src/Stratis.Bitcoin.IntegrationTests.Common/Runners/BitcoinCoreRunner.cs

```diff
@@ -53,6 +54,15 @@ public override void Start()
             string logMode = Debugger.IsAttached ? "-debug=net" : string.Empty;
             TimeSpan duration = TimeSpan.FromSeconds(15);
 
+            // The complete path bitcoind uses to locate (e.g.) the block files consists of the source code build folder,
+            // the relative path within the test case folders, and the bitcoind network-specific path to its block database
+            // This adds roughly 37 characters onto the full data folder path: \regtest\blocks\index/MANIFEST-000001
+
+            // By throwing here we avoid a pointless 5-minute wait for bitcoind to start up (it will 'start' and then soon
+            // crash, which results in the getblockhash RPC call timing out later on in the startup sequence).
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && (new FileInfo(this.DataFolder).FullName.Length > 222))
```

Comment:
Magic number - 222. Probably better as a constant.

Reply:
I'm not sure it's even worth it, this check is only added to 'fail fast' in case the developer isn't aware that having paths that are too long causes bitcoind to not start up properly.